rasa/core/channels/slack.py

```diff
@@ -310,6 +328,25 @@ def _get_interactive_response(action: Dict) -> Optional[Text]:
 
         return response.text("")
 
+    def get_metadata(self, request: Request) -> Dict[Text, Any]:
+        """Extracts the metadata from a slack API event (https://api.slack.com/types/event).
+
+        Args:
+            request: A `Request` object that contains a slack API event in the body.
+
+        Returns:
+            Metadata extracted from the sent event payload. This includes the output channel for the response, the text
+            from the user and users that have installed the bot.
+        """
+        slack_event = request.json
+        event = slack_event.get("event", {})
+        return {
+            "out_channel": event.get("channel"),
+            "text": event.get("text"),
```

Comment:
it feels a bit weird that the `metadata` also returns `sender` and `text`. The metadata is meant to provide additional information around a message. `text` and `sender` are the message. So in my opinion it would be a bit cleaner to keep this separated, what you think?